### DISCUSSION TITLE
Edge Classification

### DISCUSSION THREAD

**Discussion by PrashnikB** [pyg-team/pytorch_geometric#8337] [Q&A]
I am trying to make a GNN model for credit card fraud detection.
The graph that I built has users as nodes and the transactions as edges I want to label each edge as fraud or not but I am not able to find any example where edge label has been been predicted. I am not trying to do link prediction, I aldready have the edges I only want to classify them as fraud or not fraud.
Can someone help

**Comment by xnuohz** [comment]
https://github.com/pyg-team/pytorch_geometric/blob/master/examples/link_pred.py

**Reply by PrashnikB** [reply]
In this example they are doing link prediction but I aldready have the links I want to do edge classification.

**Reply by PrashnikB** [reply]
Is there anyway to use link prediction for edge classification?

**Reply by xnuohz** [reply]
IMO, link prediction is similar to edge classification.
In edge classification, you could first use GNN extract `node` and `edge` features, then define a downstream function like `f(node_src_feature, edge_feature, node_dst_feature)` to predict the edge's label.
